internal/json/schema.go

```diff
@@ -0,0 +1,56 @@
+/*
+ * @license
+ * Copyright 2023 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package json
+
+import (
+	"fmt"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/internal/log"
+	"github.com/invopop/jsonschema"
+	"github.com/spf13/afero"
+	"path/filepath"
+)
+
+func CreateJSONSchemaFile(value interface{}, fs afero.Fs, folderPath, filename string) error {
```

Comment:
I would suggest to return a generated JSON schema as as byte array (or stream). What to do with the schema (e.g. write to the file) is responiibliti of the caller.